01-syntax/06-project-elife/elife.js

```diff
@@ -1 +1,361 @@
+
+actionTypes.grow = function (critter) {
+  var crit = critter;
+  crit.energy += 0.5;
+  return true;
+};
+
+actionTypes.move = function (critter, vector, action) {
+  var crit = critter;
+  var dest = this.checkDestination(action, vector);
+  if (dest === null ||
+    crit.energy <= 1 ||
+    this.grid.get(dest) !== null) return false;
+  crit.energy -= 1;
+  this.grid.set(vector, null);
+  this.grid.set(dest, crit);
+  return true;
+};
+
+actionTypes.eat = function (critter, vector, action) {
+  var crit = critter;
+  var dest = this.checkDestination(action, vector);
+  var atDest = dest !== null && this.grid.get(dest);
+  if (!atDest || atDest.energy === null) return false;
+  crit.energy += atDest.energy;
+  this.grid.set(dest, null);
+  return true;
+};
+
+actionTypes.kill = function (critter, vector) {
+  var dest;
+  var action = critter.act(new View(this, vector));
+  if (action && action.type === 'kill') {
+    dest = this.checkDestination(action, vector);
+    if (dest) {
+      this.grid.set(vector, null);
+      this.grid.set(dest, critter);
+    }
+  }
+};
+
+actionTypes.reproduce = function (critter, vector, action) {
+  var crit = critter;
+  var baby = getElementFromChar(this.legend,
+    crit.originChar);
+  var dest = this.checkDestination(action, vector);
+  if (dest === null ||
+    crit.energy <= 2 * baby.energy ||
+    this.grid.get(dest) !== null) return false;
+  crit.energy -= 2 * baby.energy;
+  this.grid.set(dest, baby);
+  return true;
+};
+
+function Plant() {
+  this.energy = 3 + Math.random() * 4;
+}
+Plant.prototype.act = function (view) {
+  var space;
+  if (this.energy > 15) {
+    space = view.find(' ');
+    if (space) return { type: 'reproduce', direction: space };
+  }
+  if (this.energy < 20) return { type: 'grow' };
+  return null;
+};
+
+function PlantEater() {
+  this.energy = 20;
+  this.lastMove = 's';
+}
+PlantEater.prototype.act = function (view) {
+  var space = view.find(' ');
+  var plant = view.find('*');
+  var plantsAmount = view.findAll('*');
+  if (this.energy > 100 && space) {
+    return { type: 'reproduce', direction: space };
+  }
+  if (plantsAmount.length > 1) {
+    if (plant) {
+      this.lastMove = plant;
+      return { type: 'eat', direction: this.lastMove };
+    }
+  }
+  if (view.look(this.lastMove) === '#' || view.look(this.lastMove) === '0') {
+    if (space) {
+      this.lastMove = space;
+      return { type: 'move', direction: this.lastMove };
+    }
+  }
+  return { type: 'move', direction: this.lastMove };
+};
+
+LifelikeWorld.prototype.letAct = function (critter, vector) {
+  var crit = critter;
+  var action = crit.act(new View(this, vector));
+  var handled = action && action.type in actionTypes &&
+      actionTypes[action.type].call(this, crit, vector, action);
+  if (!handled) {
+    crit.energy -= 0.2;
+    if (crit.energy <= 0) {
+      this.grid.set(vector, null);
+    }
+  }
+};
+
+window.valley = new LifelikeWorld(plan, { '#': Wall,
```

Comment:
Actually, it maks sense to go without `valley` variable, you can initialize `LifelikeWorld ` directly in 
https://github.com/DeuceX/sf-eng-init-gl-spring-2016/blob/ELife/01-syntax/06-project-elife/index.html#L8